Added Syntax Highlighting support for Notepad++

ramb0x opened the pull request:
I added in formatting for all the regex criteria from the sublime file  and from the documentation, including hex prefixes and numerical suffixes.
As for the style, I used the clean highlighting scheme in code from "Young Person's Guide to Bond" page:
https://microsoft.github.io/bond/manual/bond_cpp.html#idl-syntax

Let me know if I'm missing anything.

sapek commented on tools/syntax/Notepad/bond.xml:
```diff
@@ -0,0 +1,64 @@
+            <Keywords name="Folders in comment, open"></Keywords>
+            <Keywords name="Folders in comment, middle"></Keywords>
+            <Keywords name="Folders in comment, close"></Keywords>
+            <Keywords name="Keywords1">int8&#x000D;&#x000A;uint8&#x000D;&#x000A;int16&#x000D;&#x000A;uint16&#x000D;&#x000A;int32&#x000D;&#x000A;uint32&#x000D;&#x000A;bool&#x000D;&#x000A;float&#x000D;&#x000A;double&#x000D;&#x000A;string&#x000D;&#x000A;wstring&#x000D;&#x000A;&#x000D;&#x000A;&#x000D;&#x000A;&#x000D;&#x000A;&#x000D;&#x000A;</Keywords>
```
Also: int64, uint64, blob

sapek commented on tools/syntax/Notepad/bond.xml:
```diff
@@ -0,0 +1,64 @@
+<NotepadPlus>
+    <UserLang name="Bond" ext="bond" udlVersion="2.1">
+        <Settings>
+            <Global caseIgnored="yes" allowFoldOfComments="yes" foldCompact="no" forcePureLC="0" decimalSeparator="2" />
```
caseIgnored="no"

sapek commented on tools/syntax/Notepad/bond.xml:
```diff
@@ -0,0 +1,64 @@
+            <Keywords name="Numbers, prefix2">0x</Keywords>
+            <Keywords name="Numbers, extras1">A B C D E F a b c d e f </Keywords>
+            <Keywords name="Numbers, extras2"></Keywords>
+            <Keywords name="Numbers, suffix1">L l UL ul u U F f ll LL ull ULL</Keywords>
```
Bond doesn't use number sufixes

sapek commented on tools/syntax/Notepad/bond.xml:
```diff
@@ -0,0 +1,64 @@
+            <Keywords name="Keywords3">true&#x000D;&#x000A;false&#x000D;&#x000A;nothing</Keywords>
+            <Keywords name="Keywords4">vector&#x000D;&#x000A;map&#x000D;&#x000A;list&#x000D;&#x000A;nullable&#x000D;&#x000A;set</Keywords>
+            <Keywords name="Keywords5">a&#x000D;&#x000A;b&#x000D;&#x000A;c&#x000D;&#x000A;d&#x000D;&#x000A;e&#x000D;&#x000A;f&#x000D;&#x000A;g&#x000D;&#x000A;h&#x000D;&#x000A;i&#x000D;&#x000A;j&#x000D;&#x000A;k&#x000D;&#x000A;l&#x000D;&#x000A;m&#x000D;&#x000A;n&#x000D;&#x000A;o&#x000D;&#x000A;p&#x000D;&#x000A;q&#x000D;&#x000A;r&#x000D;&#x000A;s&#x000D;&#x000A;t&#x000D;&#x000A;u&#x000D;&#x000A;v&# ...
+            <Keywords name="Keywords6"></Keywords>
```
required, optional, required_optional

sapek commented:
Could you please rebase squash into a single commit and change the commit title to `Add syntax highlighting support for Notepad++` (for consistency).

One thing that doesn't work is map keys don't get highlighted. I don't know N++ lang syntax so I don't know why.

ramb0x commented:
@sapek
Done! Yeah the process is a bit convoluted.
Added in missing keyboards, removed number suffixes and ignorecase, map should be highlighting now as well.

sapek commented:
Thanks. However something went wrong in the rebase. Your commit now has changes in two files that were modified in other commits: appveyor.yml and cs/src/io/unsafe/StreamCloning.cs.

ramb0x commented:
@sapek 
Ah, got it all figured out now. Good to go.

sapek commented:
Thanks. Merged manually @4bea59d783 (renamed `Notepad` to `Notepa-plus-plus`).
